devtools/dump_devinfo.py

```diff
@@ -237,22 +306,40 @@ async def get_smart_fixture(device: SmartDevice):
             )
             exit(1)
         except Exception as ex:
-            click.echo(click.style(f"FAIL {ex}", fg="red"))
+            if (
+                not test_call.should_succeed
+                and hasattr(ex, "error_code")
+                and ex.error_code == SmartErrorCode.UNKNOWN_METHOD_ERROR
+            ):
+                click.echo(click.style("FAIL - EXPECTED", fg="green"))
+            else:
+                click.echo(click.style(f"FAIL {ex}", fg="red"))
         else:
             if not response:
-                click.echo(click.style("FAIL not suported", fg="red"))
+                click.echo(click.style("FAIL no response", fg="red"))
             else:
-                click.echo(click.style("OK", fg="green"))
+                if not test_call.should_succeed:
+                    click.echo(click.style("OK - EXPECTED FAIL", fg="red"))
+                else:
+                    click.echo(click.style("OK", fg="green"))
                 successes.append(test_call)
 
     requests = []
     for succ in successes:
-        requests.append({"method": succ.method})
-
-    final_query = {"multipleRequest": {"requests": requests}}
+        requests.append(succ.request)
 
+    final = {}
     try:
-        responses = await device.protocol.query(final_query)
+        end = len(requests)
+        step = 10  # Break the requests down as there seems to be a size limit
```

Comment:
We introduced some API in #502 to help with such issues, maybe we should implement this for tapo* classes, too? I doubt this is necessary for the 0.6.0 release, but something to keep in mind to look at some point.